hazelcast/src/main/java/com/hazelcast/map/impl/PartitionContainer.java

```diff
@@ -225,4 +239,29 @@ public void setLastCleanupTimeCopy(long lastCleanupTimeCopy) {
         this.lastCleanupTimeCopy = lastCleanupTimeCopy;
     }
 
+    // -------------------------------------------------------------------
+    // WATCH OUT: never use directly! use MapContainer.getIndex() instead
+    // -------------------------------------------------------------------
+    Indexes getIndexes(String name) {
+        final boolean globalIndex = false;
+        Indexes ixs = indexes.get(name);
+        if (ixs == null) {
+
+            MapServiceContext mapServiceContext = mapService.getMapServiceContext();
+            MapContainer mapContainer = mapServiceContext.getMapContainer(name);
+
+            InternalSerializationService ss = (InternalSerializationService)
+                    mapServiceContext.getNodeEngine().getSerializationService();
+            Extractors extractors = mapServiceContext.getMapContainer(name).getExtractors();
+            IndexProvider indexProvider = mapServiceContext.getIndexProvider(mapContainer.getMapConfig());
+            Indexes indexesForMap = new Indexes(ss, indexProvider, extractors, globalIndex);
+            ixs = indexes.putIfAbsent(name, indexesForMap);
+            if (ixs == null) {
```

Comment:
Right now not, since the partitioned-index is used in off-heap only (which uses partition-threading). but the implementation is more generic, and initially in this PR the on-heap indexes were partitioned too. If it wasn't for performance overhead it would stay like this. I would prefer to keep the implementation a bit more generic, since the overhead is zero (it's just the first access), but in this way it may be extended later.